src/Mod/PartDesign/App/FeatureHole.cpp

```diff
@@ -2169,33 +2174,65 @@ TopoShape Hole::findHoles(std::vector<TopoShape> &holes,
-        Handle(Geom_Curve) c = BRep_Tool::Curve(edge, c_start, c_end);
-
-        // Circle?
-        if (c->DynamicType() != STANDARD_TYPE(Geom_Circle))
-            continue;
-
-        Handle(Geom_Circle) circle = Handle(Geom_Circle)::DownCast(c);
-        gp_Pnt loc = circle->Axis().Location();
-
-
+    auto add_hole = [&](Part::TopoShape const& baseshape, gp_Pnt loc) {
         gp_Trsf localSketchTransformation;
         localSketchTransformation.SetTranslation( gp_Pnt( 0, 0, 0 ),
                                                   gp_Pnt(loc.X(), loc.Y(), loc.Z()) );
 
         Part::ShapeMapper mapper;
-        mapper.populate(Part::MappingStatus::Modified, profileEdge, TopoShape(protoHole).getSubTopoShapes(TopAbs_FACE));
+        mapper.populate(Part::MappingStatus::Modified, baseshape, TopoShape(protoHole).getSubTopoShapes(TopAbs_FACE));
 
         TopoShape hole(-getID());
-        hole.makeShapeWithElementMap(protoHole, mapper, {profileEdge});
+        hole.makeShapeWithElementMap(protoHole, mapper, {baseshape});
 
         // transform and generate element map.
         hole = hole.makeElementTransform(localSketchTransformation);
         holes.push_back(hole);
+    };
+
+    int baseProfileType = BaseProfileType.getValue();
+
+    // Iterate over edges and filter out non-circle/non-arc types
+    if (baseProfileType & BaseProfileTypeOptions::OnCircles || 
+       baseProfileType & BaseProfileTypeOptions::OnArcs) {
+        for (const auto &profileEdge : profileshape.getSubTopoShapes(TopAbs_EDGE)) {
+            Standard_Real c_start;
+            Standard_Real c_end;
+            TopoDS_Edge edge = TopoDS::Edge(profileEdge.getShape());
+            Handle(Geom_Curve) c = BRep_Tool::Curve(edge, c_start, c_end);
+
+            // Circle base?
+            if (c->DynamicType() != STANDARD_TYPE(Geom_Circle)) {
+                continue;
+            }
+
+            bool closed = ( std::fabs(c_start) < Precision::Angular() && 
+                            std::fabs(c_end - 2 * std::numbers::pi) < Precision::Angular());
+
+            // Filter for circles
+            if (!(baseProfileType & BaseProfileTypeOptions::OnCircles) && closed) {
+                continue;
+            }
+            
+            // Filter for arcs
+            if (!(baseProfileType & BaseProfileTypeOptions::OnArcs) && !closed) {
+                continue;
+            }
+
+            Handle(Geom_Circle) circle = Handle(Geom_Circle)::DownCast(c);
+            add_hole(profileEdge, circle->Axis().Location());
+        }
+    }
+
+    // To avoid breaking older files which where not made with
+    // holes on points
+    if (baseProfileType & BaseProfileTypeOptions::OnPoints) {
+        // Iterate over vertices while avoiding edges so that curve handles are ignored
+        for (const auto &profileVertex : profileshape.getSubTopoShapes(TopAbs_VERTEX, TopAbs_EDGE)) {
+            TopoDS_Vertex vertex = TopoDS::Vertex(TopoDS_Shape(profileVertex.getShape()));

```

Comment:
```suggestion
            TopoDS_Vertex vertex = TopoDS::Vertex(profileVertex.getShape());
```

I think that this should work fine too?